tests/unit/backend/service/test_CalibrationService.py

```diff
@@ -287,8 +316,68 @@ def test_assessQuality(
         result = self.instance.assessQuality(mockRequest)
 
         # Assert the result is as expected
-        expected_record = "mock_calibration_record"
-        assert result == expected_record
+        expected_record_mockId = "mock_calibration_record"
+        assert result.mockId == expected_record_mockId
+
+        # Assert expected calibration metric workspaces have been generated
+        ws_name_stem = "57514_calibrationMetrics_ts123"
+        for ws_name in [ws_name_stem + "_sigma", ws_name_stem + "_strain"]:
+            assert self.instance.dataFactoryService.doesWorkspaceExist(ws_name)
+
+    @patch(thisService + "CalibrationMetricsWorkspaceIngredients", return_value=MagicMock())
+    def test_readQuality_mock_metrics_workspace_ingredients(
+        self, mockCalibrationMetricsWorkspaceIngredients  # noqa: ARG002
+    ):
+        run = MagicMock()
+        version = MagicMock()
+        calibRecord = CalibrationRecord.parse_raw(Resource.read("inputs/calibration/CalibrationRecord.json"))
+        self.instance.dataFactoryService.getCalibrationRecord = MagicMock(return_value=calibRecord)
+        with pytest.raises(Exception) as excinfo:  # noqa: PT011
+            self.instance.readQuality(run, version)
+        assert "The input table is empty" in str(excinfo.value)
```

Comment:
The test name should reflect that it is expecting an Exception. (negative case)

Reply:
I've changed the name of that test. I've also split `test_readQuality` into 2 tests. It actually had 3 tests (i.e. 3 `readQuality` calls), of which 2 were testing for an exception. However, upon closer examination, one of the expected exceptions was from `pydantic` validation, so I removed that test as unnecessary.